lib/ExternalModule.js

```diff
@@ -116,6 +116,12 @@ class ExternalModule extends Module {
 	size() {
 		return 42;
 	}
+
+	updateHash(hash) {
+		hash.update(this.type);
```

Comment:
Could you include `` `${this.id}` `` and `JSON.stringify(this.optional)` into the hash?

Reply:
Since `ExternalModule` is a subclass of `Module` and it a Module already does this:

```js
hash.update(this.id + "" + this.used);
```

And `super.updateHash(hash);` is already called. Should I still include `this.id`?

Reply:
@sokra I added an `ExternalModule.test` case to verify that `this.id` is hashed (without changing implementation).

Reply:
ok great.